src/app/json.cpp

```diff
@@ -28,18 +28,61 @@
 
 /**
  * @file
- *   The file implements the config json parser;
+ *   This file defines the json parser of Network Data and Commissioner configuration.
+ *
  */
 
 #include "json.hpp"
 
+#include <exception>
 #include <iostream>
 
 #include <nlohmann/json.hpp>
 
-#include "commissioner_app.hpp"
 #include <utils.hpp>
 
+#include "commissioner_app.hpp"
+#include "file_logger.hpp"
+#include "file_util.hpp"
+
+namespace ot {
+
+namespace commissioner {
+
+/**
+ * This Exception class represents Network Data
+ * and Configuration conversion errors from JSON file.
+ *
+ * We need this because the nlohmann::json library uses
+ * exception exclusively.
+ *
+ */
+class JsonException : public std::invalid_argument
+{
+public:
+    explicit JsonException(const std::string &what_arg)
+        : std::invalid_argument(what_arg)
+    {
+    }
+    explicit JsonException(const char *what_arg)
+        : std::invalid_argument(what_arg)
+    {
+    }
+};
+
+} // namespace commissioner
+
+} // namespace ot
+
+#define SuccessOrThrow(aError)                                                                  \
+    do                                                                                          \
+    {                                                                                           \
+        if (aError != ::ot::commissioner::Error::kNone)                                         \
+        {                                                                                       \
+            throw ::ot::commissioner::JsonException(::ot::commissioner::ErrorToString(aError)); \
```

Comment:
Google C++ style guide strongly recommends against the use of exceptions. Why do we use them here?

Reply:
Because the [json](https://github.com/nlohmann/json) library uses exception exclusively. The macro `SuccessOrThrow` is only defined in this `json.cpp` file and only called in those converters (`from_json()` & `to_json()`) called by the json library. see also the doc string of the `JsonException` class above.
 
All APIs in `json.hpp` are guaranteed never throw a exception.

Reply:
Makes sense. Thanks for the clarification!